src/components/settings/FallbackHandler/index.tsx

```diff
@@ -35,7 +36,8 @@ export const FallbackHandler = (): ReactElement | null => {
 
   const hasFallbackHandler = !!safe.fallbackHandler
   const isOfficial =
-    hasFallbackHandler && safe.fallbackHandler?.value === fallbackHandlerDeployment?.networkAddresses[safe.chainId]
+    safe.fallbackHandler &&
+    fallbackHandlerDeployments?.networkAddresses[safe.chainId].includes(safe.fallbackHandler.value)
   const isTWAPFallbackHandler = safe.fallbackHandler?.value === TWAP_FALLBACK_HANDLER
```

Comment:
I know you did not introduce this. But this check is too simplified. There are many chains where no Extensible Fallback handler is deployed e.g. Base. 
Maybe we should only allow this on Mainnet, Gnosis Chain, Sepolia and Arbitrum for now as those are the chains where CoW exists?

Reply:
Wouldn't `isTWAPFallbackHandler` be false on those other chains where its not deployed which is the expected behaviour?

Reply:
But it just compares an hard coded address:
```ts
export const TWAP_FALLBACK_HANDLER = '0x2f55e8b20D0B9FEFA187AA7d00B6Cbe563605bF5'
```

There are no checks that this address is a contract on a chain so this will always be true when you replay a twap Safe for instance in case someone sets one up with the twap fallback handler.

Reply:
These are the networks where that fallback handler is deployed:
https://github.com/cowprotocol/composable-cow/blob/main/networks.json

Reply:
Ah I see. Then lets harden the condition by also checking that the chainId matches one of the deployed networks. I hard-coded the list of those networks provided by the link for now.

Reply:
Also added some tests for the twap fallback handler message